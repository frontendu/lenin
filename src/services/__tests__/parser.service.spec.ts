import { ParserService } from 'services/parser.service';

import blog from './fixtures/blog';
import twitter from './fixtures/twitter';
import yandex from './fixtures/yandex';

class StubSiteService {
  constructor(private readonly content) {}

  async getContent() {
    return this.content;
  }
}

describe('ParserService', () => {
  [[blog, 'blog'], [twitter, 'twitter'], [yandex, 'yandex']].forEach(
    ([content, name]) => {
      it(`should properly renders ${name}`, async () => {
        expect(
          await new ParserService(new StubSiteService(content)).getSiteTitle(
            name
          )
        ).toMatchSnapshot();
      });
    }
  );
});
