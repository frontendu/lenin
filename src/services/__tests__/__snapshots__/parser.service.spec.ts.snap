// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParserService should properly renders blog 1`] = `"CO2 emissions on the web | Danny van Kooten"`;

exports[`ParserService should properly renders devto 1`] = `"A beginner’s journey into software development in Nigeria. - DEV Community"`;

exports[`ParserService should properly renders twitter 1`] = `
"Андрей Ситник в Твиттере: «Для разработчиков npm-пакетов, которые хотят выпускать двойные ESM/CommonJS модули, я сделал  инструмент dual-publish.

https://t.co/1Iha0Fy50S

Он учитывает все особенности как ESM в сборщиках, так и в ESM в Node.js и в браузерах.… https://t.co/XUd6fSs0AQ»"
`;

exports[`ParserService should properly renders yandex 1`] = `"Эрдоган и Путин обсудят ситуацию в Идлибе: Яндекс.Новости"`;
